### DISCUSSION TITLE
Updating to 1.28.2 broke my connection to Grafana due to SSL

### DISCUSSION THREAD

**Discussion by ijord** [teslamate-org/teslamate#3510] [General]
I run dockers on a Synology. All working well for a year. This morning updated to 1.28.2 via the teslamate images. This updated both the teslamate and grafana images.

Teslamate runs fine.

I had to add a new folder for grafana-plugins that I never needed before. Without this grafana would just crash with a mkdir error.

Once that is added, grafana runs, but all the dashboards are blank. There is a query error: "Status: 500. Message: db query error: pq: SSL is not enabled on the server"

Any ideas? I can't find any easy references of how to turn on SSL in postgres or disable the need for SSL in grafana. This is purely local instance with no exposure to the internet, so I don't care about SSL.

**Comment by eliluong** [comment]
Just a guess, but you can try adding `GF_DATABASE_SSL_MODE=disable` to environment variables for the grafana container.

**Comment by ijord** [comment]
I tried GF_DATABASE_SSL_MODE=disable to the grafana container with no success. Also tried sslmode = disable, and added this to the DB name, I even edited the postgresql.conf to change "SSL" from off to on. None of this helped.

Grafna for sure is talking to postgres, if I shut down the postgres container I get a missing server connection error.

**Comment by ijord** [comment]
Ok, so I tried previous images. 1.28.0. 1.28.1, and 1.28.2 all work fine. Only "latest" which is 1.28 causes the SSL error.

..But then I made a fresh "latest" grafana docker and it all worked, using the same image, but a new instance in Synology's docker manager. I think this has something to do with the fact that I have all of teslamate set up as a project and the containers are all managed in one group. 

When I did a "build" on the docker compose, it fixed it, even running "latest." Not enough of an expert to know why I needed to do that, I have not had to before.

**Reply by ftt1211** [reply]
What do you mean doing a "build?" I'm having the same issue after updating.

**Comment by cwanja** [comment]
Wonder if this is related to #3498 - @simonsmh ?

**Reply by simonsmh** [reply]
Have you tried setting DATABASE_SSL_MODE=require which turn the pg ssl connection on? The cause might be that I changed the default pg version from 10.0 to 15.0.

**Comment by ijord** [comment]
I don't have the problem anymore and I don't know how to re-create it. I had the problem when I just updated the image for docker, but the ultimate fix was to do a "build" of the teslamate package after updating the image. I've never had to do this before, but once I did it everything worked fine. It does appear to be tied to #3498, but I'm not sure what the build process does to resolve it.

**Reply by ironbract** [reply]
Could you provide a link or explanation of how to "build" the Teslamate Docker package? I don't think I've needed to do that since I started with Teslamate several years ago, and don't know the intricacies of Docker. Thank you!

**Comment by dopgelder** [comment]
Ran into the same issue. On my Synology docker configuration page, did a “reset” of the teslamate-GRAFANA container and that fixed the issue.

**Reply by ironbract** [reply]
By reset, do you mean stopping, removing and reinstalling the grafana container?

**Reply by dopgelder** [reply]
In the Synology GUI, click on container manager, the teslamate-GRAFANA container and shut it down. In actions menu you can select “reset” and after  you have updated the container to the new image, it fixed the problem for me.

**Reply by ftt1211** [reply]
I can't seem to reset the container. I get the message: "Unable to perform this operation, possibly because the network connection is unstable or the system is busy. Please try again later."

**Reply by Toasterme** [reply]
@dopgelder you're a legend !!! i got the same issue with Synology too!  and the STOP teslamate-GRAFANA container + reset + start did the trick !  i had the error below:
![image](https://github.com/teslamate-org/teslamate/assets/732881/edf71e81-6d57-4fe3-a28c-15d8ab11a8d6)
then if i clicked the Pink mark the error i had was : 
Status: 500. Message: db query error: pq: SSL is not enabled on the server

But nothing showed up in the log...

now fixed, thanks to YOU !

**Comment by lroths** [comment]
put the parameter   DATABASE_SSL_MODE=disable   in the container grafana

it is not GF_...  as specified in a previous comment

**Reply by BelayFR** [reply]
@lroths => Thx for the detailed parameter syntax.... Worked for me like a charm !
